llvm/lib/Target/AMDGPU/AMDGPUAlwaysInlinePass.cpp

```diff
@@ -98,6 +99,7 @@ static bool alwaysInlineImpl(Module &M, bool GlobalOpt) {
       if (TT.getArch() == Triple::amdgcn &&
           A.getLinkage() != GlobalValue::InternalLinkage)
         continue;
+      Preserved = false;
```

Comment:
The normal way to do this is track changed not preserved

Reply:
ah, thank you!